pymodbus/repl/server/main.py

```diff
@@ -16,38 +17,86 @@
 
 CANCELLED_ERROR = asyncio.exceptions.CancelledError
 
+_logger = logging.getLogger(__name__)
 
-@click.group("ReactiveModbusServer")
-@click.option("--host", default="localhost", help="Host address")
-@click.option("--web-port", default=8080, help="Web app port")
-@click.option(
-    "--broadcast-support",
-    is_flag=True,
-    default=False,
-    help="Support broadcast messages",
-)
-@click.option(
-    "--repl/--no-repl",
-    is_flag=True,
-    default=True,
-    help="Enable/Disable repl for server",
-)
-@click.option(
-    "--verbose", is_flag=True, help="Run with debug logs enabled for pymodbus"
-)
-@click.pass_context
-def server(ctx, host, web_port, broadcast_support, repl, verbose):
+CONTEXT_SETTING = {"allow_extra_args": True, "ignore_unknown_options": True}
+
+# TBD class ModbusServerConfig:
+
+
+class ModbusServerTypes(str, Enum):
+    """Server types."""
+
+    # ["tcp", "serial", "tls", "udp"]
+    tcp = "tcp"  # pylint: disable=invalid-name
+    serial = "serial"  # pylint: disable=invalid-name
+    tls = "tls"  # pylint: disable=invalid-name
+    udp = "udp"  # pylint: disable=invalid-name
+
+
+class ModbusFramerTypes(str, Enum):
+    """Framer types."""
+
+    # ["socket", "rtu", "tls", "ascii", "binary"]
+    socket = "socket"  # pylint: disable=invalid-name
+    rtu = "rtu"  # pylint: disable=invalid-name
+    tls = "tls"  # pylint: disable=invalid-name
+    ascii = "ascii"  # pylint: disable=invalid-name
+    binary = "binary"  # pylint: disable=invalid-name
+
+
+def _completer(incomplete: str, valid_values: list[str]) -> list[str]:
```

Comment:
Same here `list[str]` will raise TypeError, needs to be `typing.List`

Reply:
nope